### DISCUSSION TITLE
How to remove old records

### DISCUSSION THREAD

**Discussion by pietru2004** [pocketbase/pocketbase#4042] [Q&A] [answered]
I want to ask how to make code to handle time based tokens,
I mean remove records older than 5/10 minutes?

**Comment by ganigeorgiev** [comment, marked as answer]
It is not clear what do you mean by _"handle time based tokens"_.

In general for deleting records periodically, you can register a cron job with the frequency interval you want and perform the delete operation from there.

I'd assume that you use the JS hooks, so here is one possible example (_I haven't tested it_):

```js
// pb_hooks/main.pb.js

// runs on every 10 minutes (if you want to be more precise run on every minute, aka. * * * * *)
cronAdd("delete_old_records", "*/10 * * * *", () => {
    // now - 10min
    let minDate = new Date(Date.now() - 1000 * 60 * 10);
    minDate = minDate.toISOString().replace("T", " "); // convert to rfc3339 format

    // use the builtin helper OR
    // alternatively you can use the query builder to construct your query
    // https://pocketbase.io/docs/js-records/#custom-record-query
    const records = $app.dao().findRecordsByFilter(
        "yourCollection",
        "created <= {:created}",
        0, // no limit, aka. retrieve all records matching the criteria
        0, // no offset
        {
            "created": minDate,
        }
    )
    
    // delete as 1 transaction
    // the transaction is commited ONLY if all delete operations succeed;
    // if you don't want this behavior, remove the transaction and call directly `$app.dao().deleteRecord()`)
    $app.dao().runInTransaction((txDao) => {
        for (let record of records) {
            txDao.deleteRecord(record)
        }
    })
})
```

If the above doesn't help, please provide some details on what you are trying to do.

**Reply by pietru2004** [reply]
basicaly I want to gen random token to aditionaly authorize requests and I want it's validity to be like 5/10 min

**Reply by ganigeorgiev** [reply]
Hm, but this has nothing to do with removing records, or at least I don't understand how it is related. Note that by default PocketBase doesn't store the auth tokens in the database.

If you want to generate an auth token with duration different then the one configured from the Admin UI, you can invoke manually the [`$security.createJWT()`](https://pocketbase.io/jsvm/functions/_security.createJWT.html) helper:

```js
const record = ...

const token = $security.createJWT(
    {
        "id":           record.id,
        "type":         "authRecord",
        "collectionId": record.collection().id,
    },
    (record.tokenKey() + $app.settings().recordAuthToken.secret),
    100 // duration in seconds
)
```

**Reply by pietru2004** [reply]
I am making web app that needs other app to be able to interact with db, users will require auth, but I want to implement additional token as special auth that will be nonexistant in db after some time
Basicaly I am making app that will log battles of users that are authorized with system.

**Reply by pietru2004** [reply]
and webapp is gona be thing to display matches, while other app will be reporting finished games
